### ISSUE TITLE
Paloalto generator creates incorrect application/service when destination_ports are not specified for TCP/UDP

### ISSUE THREAD

**Issue by abhindes** [google/capirca#270]
Currently, when a protocol (say TCP/UDP) is specified, but no ports are specified, it creates 'application-default' as the application. 'application-default' is to be used for custom applications being created (such as services with ICMP types) only, and not for TCP/UDP with any port.

**Expected behavior** -
1. When destination_ports are not specified for TCP/UDP, it should create a service entry.
2. The port for the new service entry created should be "any". "any' is supported in PANOS. It is chosen over 1-65535 to enhance readability/auditing.
3. [Best practice] The service entry created can be re-used if it's created as "tcp-any" and "udp-any". The sub-optimal alternative is to create a new service entry for each rule with TCP port "any" (just like it is created for specific TCP ports currently).  

**Known caveat** - port "any" in PANOS internally implies port 0-65535, but port 0 usage is discouraged due to its often undefined behavior. 

**To be verified prior to implementation** - If port "any" is specified in the service entry, and when the configuration is displayed, does it remain as "any", or does it get expanded to 0-65535. If it remains as "any", it should be implemented as specified earlier. If the displayed config (due to PANOS internal parser) expands it to 0-65535, then it should be implemented as the entire port range, to comply with [this requirement](https://github.com/google/capirca/blob/master/doc/generator_patterns.md#syntax-of-the-config-from-the-generator-and-on-device-should-match-when-cryptographically-verified).

Snippet from ./sample_stateful_multitarget_simple.xml (output from the sample_stateful_multitarget_simple.pol file)

What is currently happening -
```
               <entry name="test-tcp-log-both">
                  <description>Testing log-both for tcp.</description>
                  <to>
                    <member>external</member>
                  </to>
                  <from>
                    <member>internal</member>
                  </from>
                  <source>
                    <member>any</member>
                  </source>
                  <destination>
                    <member>any</member>
                  </destination>
                  <service>
                    <member>application-default</member>   <---- incorrect
                  </service>
                  <action>allow</action>
                  <rule-type>interzone</rule-type>
                  <application>
                    <member>any</member>
                  </application>
                  <tag>
                    <member>internal_external_policy-comment-1</member>
                  </tag>
                  <log-start>yes</log-start>
                  <log-end>yes</log-end>
                </entry>

```

What should be happening - 

```
             <entry name="test-tcp-log-both">
                  <description>Testing log-both for tcp.</description>
                  <to>
                    <member>external</member>
                  </to>
                  <from>
                    <member>internal</member>
                  </from>
                  <source>
                    <member>any</member>
                  </source>
                  <destination>
                    <member>any</member>
                  </destination>
                  <service>
                    <member>tcp-any</member>   <---- this is correct
                  </service>
                  <action>allow</action>
                  <rule-type>interzone</rule-type>
                  <application>
                    <member>any</member>       
                  </application>
                  <tag>
                    <member>internal_external_policy-comment-1</member>
                  </tag>
                  <log-start>yes</log-start>
                  <log-end>yes</log-end>
                </entry>
```

Service entry for "tcp-any"-

```
            <entry name="tcp-any">
              <protocol>
                <tcp>
                  <port>any</port>
                </tcp>
              </protocol>
            </entry>
```

Note: Expected behavior when ICMP and TCP/UDP are specified together will be created as a separate issue (#272)

**Comment by kevinsteves** [comment]
Enhancement in #276

**Comment by abhindes** [comment]
Closed since #276 is merged.
